ocaml/testsuite/tests/typing-gadts/test.ml

```diff
@@ -689,9 +689,9 @@ let f : type a b. (a,b) eq -> (<m : a; ..> as 'c) -> (<m : b; ..> as 'c) =
 ;; (* fail *)
 [%%expect{|
 type (_, _) eq = Eq : ('a, 'a) eq
-Line 3, characters 18-72:
-3 | let f : type a b. (a,b) eq -> (<m : a; ..> as 'c) -> (<m : b; ..> as 'c) =
-                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Lines 3-4, characters 4-15:
+3 | ....f : type a b. (a,b) eq -> (<m : a; ..> as 'c) -> (<m : b; ..> as 'c) =
+4 |   fun Eq o -> o
```

Comment:
Note that this change, and the handful of others in this commit that have to do with location, were due to a stale parser file that is fixed in https://github.com/ocaml-flambda/flambda-backend/pull/1949/commits/9a71a4e2a2b35ee187cdf9bb9dec1ae386610d44. That commit undoes these test changes.